specs/decimal/decimal.md

```diff
@@ -45,7 +57,152 @@ data structure may be re-expressed in each language using language-specific cons
+        var yfloor = Decimal.Floor(y);
+        var xceil = Decimal.Ceiling(x);
+        var z = --yfloor * ++xceil * -Decimal.Truncate(y);
+
+        Console.WriteLine($"x = {x}");
+        Console.WriteLine($"y = {y}");
+        Console.WriteLine($"z = {z}");
+
+        Environment.Exit((int)z);
+    }
+}
+```
+
+# 5. C++
+
+This program illustrates the reference example using Windows App SDK's C++ decimal class.
+
+```c++
+#include <windows.h>
+#include <stdio.h>
+#include <decimal.h>
+
+int main()
+{
+    const Microsoft::Windows::Foundation::decimal a{ 1 };
+    const Microsoft::Windows::Foundation::decimal b{ 0.5 };
+    const Microsoft::Windows::Foundation::decimal c{ L"-6.66" };
+    const Microsoft::Windows::Foundation::decimal d{ L"1.23" };
+    const Microsoft::Windows::Foundation::decimal e{ L"4567.089" };
+    const Microsoft::Windows::Foundation::decimal f{ -4ll };
+    const Microsoft::Windows::Foundation::decimal g{ 1967u };
+    const Microsoft::Windows::Foundation::decimal h{ 1001.0f };
+
+    auto x = (((a - b) + (c % d)) * e).round(5) / f;
+    auto y = x++.round(3).clamp(f, g);
+    auto z = --y.floor() * ++x.ceil() * -y.truncate();
+
+    WEX::Logging::Log::Comment(WEX::Common::String().Format(L"x = %ls", x.to_string().c_str()));
+    WEX::Logging::Log::Comment(WEX::Common::String().Format(L"y = %ls", y.to_string().c_str()));
+    WEX::Logging::Log::Comment(WEX::Common::String().Format(L"z = %ls", z.to_string().c_str()));
+
+    printf(L"x = %ls\n", x.to_string().c_str());
+    printf(L"y = %ls\n", y.to_string().c_str());
+    printf(L"z = %ls\n", z.to_string().c_str());
+    return z.to_int32();
+}
+```
+
+# 6. WinRT
+
+This program illustrates the reference example using Windows App SDK's WinRT `DecimalValue` struct and `DecimalHelper` runtimeclass.
+
+```c++
+#include <windows.h>
+#include <stdio.h>
+#include <winrt/Microsoft.Windows.Foundation.h>
+
+int main()
+{
+    using DH = winrt::Microsoft::Windows::Foundation::DecimalHelper;
+    const auto a{ DH::FromInt32(1) };
+    const auto b{ DH::FromDouble(0.5) };
+    const auto c{ DH::FromString(L"-6.66") };
+    const auto d{ DH::FromString(L"1.23") };
+    const auto e{ DH::FromString(L"4567.089") };
+    const auto f{ DH::FromInt64(-4ll) };
+    const auto g{ DH::FromUInt16(1967u) };
+    const auto h{ DH::FromSingle(1001.0f) };
+
+    // WinRT equivalent to C++ formula
+    //
+    //   x = (((a - b) + (c % d)) * e).round(5) / f
+    //   y = x++.round(3).clamp(f, g)
+    //   z = --y.floor() * ++x.ceil() * -y.truncate()
+
+    auto x = DH::Divide(DH::Round(DH::Multiply(DH::Add(DH::Subtract(a, b), (DH::Modulo(c, d))), e), 5), f);
+    auto one = DH::FromInt32(1);
+    auto y = DH::Clamp(DH::Round(x, 3), f, g);
+    x = DH::Add(x, one);
+    auto z = DH::Multiply(DH::Multiply(DH::Subtract(DH::Floor(y), one), DH::Add(DH::Ceiling(x), one)), DH::Negate(DH::Truncate(y)));
+
+    printf(L"x = %ls\n", DH::ToString(x).c_str());
```

Comment:
>x [](http://example.com/codeflow?start=13&length=1)

Isn't this going to produce different results than above since we no longer have the side-effects of `++x` and `--y` in play?